docs/flink-queries.md

```diff
@@ -277,6 +277,58 @@ DataStream<Row> stream = env.fromSource(source, WatermarkStrategy.noWatermarks()
     "Iceberg Source as Avro GenericRecord", new GenericRecordAvroTypeInfo(avroSchema));
 ```
 
+### Emitting watermarks
+Emitting watermarks from the source itself could be beneficial for several purposes, like harnessing the
+[Flink Watermark Alignment](https://nightlies.apache.org/flink/flink-docs-release-1.18/docs/dev/datastream/event-time/generating_watermarks/#watermark-alignment)
+feature to prevent runaway readers, or providing triggers for [Flink windowing](https://nightlies.apache.org/flink/flink-docs-release-1.18/docs/dev/datastream/operators/windows/).
+
+Enable watermark generation for an `IcebergSource` by setting the `watermarkColumn`.
+The supported column types are `timestamp`, `timestamptz` and `long`.
+Timestamp columns are automatically converted to milliseconds since the Java epoch of
+1970-01-01T00:00:00Z. Use `watermarkTimeUnit` to configure the conversion for long columns.
+
+The watermarks are generated based on column metrics stored for data files and emitted once per split.
+When using watermarks for Flink watermark alignment set `read.split.open-file-cost` to prevent
+combining multiple files to a single split.
```

Comment:
expand a little bit after this.
```
The watermarks are generated based on column metrics stored for data files and emitted once per split.
If multiple smaller files with different time ranges are combined into a single split, it can increase the out-of-orderliness and extra data buffering in the Flink state. The main purpose of watermark alignment is to reduce out-of-orderliness and excess data buffering in the Flink state. Hence it is recommended to set `read.split.open-file-cost` to a very large value to prevent combining multiple smaller files into a single split.
```

Reply:
Is there a tradeoff for increasing `read.split.open-file-cost`? Would a user need to also tune the split throttling parameter to avoid memory issues?

Reply:
Applied @stevenzwu's suggestion.

@mas-chen: The tradeoff is that with higher number of splits we add memory pressure on the planning process, and also add extra CPU pressure since there will be more splits and we have to assign more of them. Added a sentence to the end of the paragraph mentioning these.